### PR TITLE
Fix removal of marker when using gmap-cluster

### LINKED ISSUE #31: Old markers don't disappear when my v-for list changes
The title says it all. I have a bunch of markers that I render using `v-for="marker in markers"` where `markers` is a computed property.

Whenever the computed property changes, the old markers don't disappear. Only new markers are added, which is not how it should be. This did work properly when I was still using the original `gmap-marker`.

After further debugging I see it has to do with `gmap-cluster`. When I don't use `gmap-cluster` anymore then it works as expected. But according to the readme, these custom markers should also work with `gmap-cluster` right?

### PR DESCRIPTION
**Pull request by Evertt**
This fixes #31 

Please merge and create a new release. I need this quite soon.

## Edit

Never mind, it's still really buggy. Can you maybe try to bugfix this?

### CODE REVIEW

File: gmap-custom-marker.vue [hunk 1]
```diff
@@ -176,6 +176,9 @@ export default {
     }
   },
   destroyed() {
+    if (this.$clusterObject) {
+      this.$clusterObject.removeMarker(this.$overlay, true);
+    }
     this.$overlay.setMap(null);
     this.$overlay = undefined;
```
Comment (amineyarman) [inline gmap-custom-marker.vue:183]:
```suggestion
       if (this.$clusterObject) {
      this.$clusterObject.removeMarker(this.$overlay, true);
    } else {
      this.$overlay.setMap(null);
      this.$overlay = undefined;
    }
```

**Review by amineyarman — CHANGES_REQUESTED** [review]

File: gmap-custom-marker.vue (hunk 1 above)
Comment (Evertt) [reply gmap-custom-marker.vue:183]: Did you test this? I know that this is how the original gmap-marker does it, but I wasn't actually convinced that this works better than what I wrote. However if you tested this and you found that it actually solves the bugginess then I'm happy to accept your suggestion.

Comment (eregnier) [reply gmap-custom-marker.vue:183]: I merged too quickly. I integrate your code by hand Aminerman.

**Comment by eregnier** [comment]
This is now released.

File: gmap-custom-marker.vue (hunk 1 above)
Comment (amineyarman) [reply gmap-custom-marker.vue:183]: @Evertt It does work this way and didn't notice any bug. But can you describe your case?
I'll publish a demo, before the end of the week, that uses vue-google-maps with this custom marker and the cluster. That way we'll have a common test ground.

**Comment by eregnier** [comment]
Hey . Checkout the github page of this project.there is a demo 
 https://eregnier.github.io/vue2-gmap-custom-marker/dist/  where I
added today some multiple cluster markers randomly generated into a list
that looks properly managed with your PR

This is a quite good exemple for your issues I think.

Le jeu. 24 oct. 2019 à 14:21, Amine Bouyarmane <notifications@github.com> a
écrit :

**Comment by Evertt** [comment]
Well that demo isn't a proper test, because it only went wrong when I was trying to _remove_ some of the markers. Can you make it so that when you click one of the bananas that the map deletes that marker?

**Comment by eregnier** [comment]
For this test you can use the add marker button to ass V icons in a vrouped place. These markers are clusterized so we should test it this way

Edit : I just tried and it look like there is an issue on  deletion. I am closer to be a le to troubleshot :)

Edit 2 : v marker aredeleted when they are clicked

**Comment by Evertt** [comment]
But the whole point is about testing whether or not removing markers is
handled well...

**Pull request merged**